hw-model/src/lib.rs

```diff
@@ -569,7 +571,11 @@ pub trait HwModel {
             self.soc_mbox().execute().write(|w| w.execute(false));
             let soc_ifc = self.soc_ifc();
             return Err(ModelError::MailboxCmdFailed(
-                soc_ifc.cptra_fw_error_non_fatal().read(),
+                if cmd == MBOX_DOWNLOAD_FIRMWARE_CMD_ID {
```

Comment:
Eh, I don't really like this sort of hard-coded logic. Can we put the error in both fw_error_fatal and fw_error_non_fatal?

Reply:
Discussed offline; filed https://github.com/chipsalliance/caliptra-sw/issues/453 to add fatal error to non-fatal registers for keeping test functions generic.